parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -1115,10 +1115,26 @@ prettyParseError ::
   String ->
   Parser.Err v ->
   Pretty ColorText
-prettyParseError s = \case
-  P.TrivialError _ (LexerError ts e) _ -> go e
+prettyParseError s e =
+  mconcat (fst <$> renderParseErrors s e) <> lexerOutput
+  where
+    lexerOutput :: Pretty (AnnotatedText a)
+    lexerOutput =
+      if showLexerOutput
+        then "\nLexer output:\n" <> fromString (L.debugLex' s)
+        else mempty
+
+renderParseErrors ::
+  forall v.
+  Var v =>
+  String ->
+  Parser.Err v ->
+  [(Pretty ColorText, [Range])]
+renderParseErrors s = \case
+  P.TrivialError _ (LexerError ts e) _ -> [(go e, ranges)]
```

Comment:
The diffs in this function look noisy, but if you turn off whitespace diffs you'll see it's mostly just re-arranging things into a let binding and adding the appropriate ranges for each error.